sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_cluster.asciidoc

```diff
@@ -0,0 +1,200 @@
+kernelObject);
+----
+
+with the defition of `clusterGroupRange` being as follows - 
+
+[%header,cols="1,5"]
+|===
+
+|`clusterGroupRange`
+|`sycl::range` defining the number of workgroups which form a cluster in each dimension
+|===
+
+[NOTE]
+====
+The total number of workgroups must be a multiple of the cluster range in each dimension.
+====
+
+If non-zero cluster dimensions are specified when launching a kernel on non-supported hardware, it should result in a runtime-error.
+
+=== Accessing the Work Group Cluster from within the kernel
+
+Building upon the group heirarchy in SYCL, this proposal another heirachy in the group after `sub_group` and `group` (for workgroups), a class to be called as `cluster_group`, which will  be accessible via the `nd_item` class, via a method to be introduced called `get_cluster_group()`
+
+[%header,cols="10,5"]
+|===
+
+|`cluster_group<Dimensions> nd_item::get_cluster_group()`
+|Returns the consituent `cluster_group` in the kernel, representing this `cluster_group`'s overall position in the `nd_range`
+|===
+
+
+The `cluster_group` class to have the following methods, to access the various id's 
+
+
+[%header,cols="5,5"]
+|===
+|Method
+|Description
+
+|`size_t get_group_linear_id(void)`
+|Returns the flattened id of the calling `group` within the cluster
+
+|`size_t get_local_linear_id(void)`
+|Returns the flattened index of the calling `workitem` within the cluster
+
+|`size_t get_group_linear_range(void)`
+|Returns the toal number of `groups` within the cluster
+
+|`size_t get_group_id(size_t)`
+|Returns the id of the calling workgroup along the specified dimension
+
+|`size_t get_group_linear_range(size_t)`
+|Returns the toal number of `groups` within the cluster in the specified dimension
+
+|`size_t get_local_linear_range(void)`
+|Returns the total number of `workItems` within the cluster
+|===
+
+
+To obain the total number of clusters within the group, and getting the cluster id, `nd_item` to have 2 more class methods, namely `get_cluster_range` and `get_cluster_id`
+
+[%header,cols="10,5"]
+|===
+
+|`size_t nd_item::get_cluster_range(size_t)`
+|Returns the total number of `cluster_group`s in the kernel across the specified dimension
+
+|`size_t nd_item::get_cluster_id(size_t)`
+|Returns the id of the cluster across the specified dimension
+|===
+
+
+CC 9.0 devices feature the ability to synchronize all the workgroups within the cluster, and to achieve that in SYCL, this proposes add an overload to the existing `sycl::group_barrier` function, accepting `cluster_group` type.
+
+[%header,cols="10,5"]
+|===
+
+|`void sycl::group_barrier(cluster_group<Dimensions> G)`
+|Synchronizes all workgroup within the cluster
+|===
+
+To prevent un-defined behaviour, in the case where these range's or id's are queried without the cluster dimensions mentioned at launch can be considered as a cluster launch with cluster dimension as 1.
+
+=== Accessing another workgroup's local memory
```

Comment:
I suggest we move this to another extension. That way we can get them approved/implemented independently.